tree/ntupleutil/v7/inc/ROOT/RNTupleImporter.hxx

```diff
@@ -114,6 +114,14 @@ public:
       virtual void Finish(std::uint64_t nbytesWritten, std::uint64_t neventsWritten) = 0;
    };
 
+   /// Used to make adjustments to the fields of the output model
+   class RFieldModifier {
```

Comment:
Any specific reason to use a virtual class rather than a callback?

Reply:
Not particularly. It felt like the easiest or clearest way for users to provide their own, possibly stateful functionality and it's not on a performance critical path.

Reply:
I would argue that passing a lambda, function pointer or (possibly stateful) functor class feels clearer and maybe a bit less verbose in simple cases, but I'm not particularly against this solution, so for me it's fine either way.

Reply:
I think I would agree that a single `std::function` feels more natural *if* we only foresee a single callback; for multiple `virtual` functions (that may be optional), a `class` provides a single interface. A `std::function` also matches the mapping function for `RNTupleModel::AddProjectedField` and the argument passed to `RFieldBase::AddReadCallback`.